style.md

```diff
@@ -1923,6 +1924,148 @@ sptr := &T{Name: "bar"}
 </td></tr>
 </tbody></table>
 
+### Initializing Maps
+
+Prefer `make(..)` for empty maps, and maps populated
+programmatically. This makes map initialization visually
+distinct from declaration, and it makes it easy to add size
+hints later if available.
+
+<table>
+<thead><tr><th>Bad</th><th>Good</th></tr></thead>
+<tbody>
+<tr><td>
+
+```go
+var (
+  // m1 is safe to read and write;
+  // m2 will panic on writes.
+  m1 = map[T1]T2{}
+  m2 map[T1]T2
+)
+```
+
+</td><td>
+
+```go
+var (
+  // m1 is safe to read and write;
+  // m2 will panic on writes.
+  m1 = make(map[T1]T2)
+  m2 map[T1]T2
+)
+```
+
+</td></tr>
+<tr><td>
+
+Declaration and initialization are visually similar.
+
+</td><td>
+
+Declaration and initialization are visually distinct.
+
+</td></tr>
+</tbody></table>
+
+Where possible, provide capacity hints when initializing
+maps with `make()`.
+
+```go
+make(map[T1]T2, hint)
+```
+
+While Go's runtime map implementation only acts upon capacity
+hints in certain situations (see
+[`runtime.makemap()`](https://golang.org/src/runtime/map.go)
+for more information), this can also be a helpful cue when
+reading source code.
+
+<table>
+<thead><tr><th>Bad</th><th>Good</th></tr></thead>
+<tbody>
+<tr><td>
+
+```go
+m := make(map[T1]T2)
+
+// later...
+
+for i := 0; i < 10000; i++ {
+  // add element to m
+)
+```
+
+</td><td>
+
+```go
+m := make(map[T1]T2, 10000)
+
+// later...
+
+for i := 0; i < 10000; i++ {
+  // add element to m
+}
+```
+
+</td></tr>
+<tr><td>
+
+The footprint of `m` is not well-understood at initialization time,
+and multiple allocations may take place as part of adding elements.
+
+</td><td>
+
+The footprint of `m` is better understood at initialization time,
+and fewer allocations are incurred when adding within its capacity.
+
+</td></tr>
+</tbody></table>
```

Comment:
@prashantv thoughts?

Reply:
I like the recommendation, but think it makes this guideline a little too long. A suggestion:
 - drop the performance-related text here
 - add a separate recommendation in the `Performance` section about specifying size hints with most of this text
 - reference that section here when we suggest the user should specify a size hint

The separate section can also be used for slices.

What do you think?

Reply:
Agreed, sounds good!